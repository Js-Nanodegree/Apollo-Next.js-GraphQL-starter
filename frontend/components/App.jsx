import React, { Component } from 'react';

import Head from 'next/head';
import Loading from './Loading';
import Navigation from './Navigation';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import withData from '../lib/withData';
import { withRouter } from 'next/router';

const App = ({ children, title, link, meta, showNavigation, Me }) => (
  <div className="fullHeight">
    <Head>
      <title>{title}</title>
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      <link
        href="https://fonts.googleapis.com/css?family=Roboto:300,400"
        rel="stylesheet"
      />
    </Head>

    {showNavigation ? <Navigation title={title} Me={Me} /> : null}

    <main id="main">{children}</main>
  </div>
);

export default App;
