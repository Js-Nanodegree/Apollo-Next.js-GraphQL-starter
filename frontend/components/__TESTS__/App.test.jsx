import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from '../App';

describe('App component', () => {
  it('should match snap shot', () => {
    const title = 'Test title';

    const component = renderer.create(
      <App title={title} showNavigation={false}>
        <div>
          <p>Test app children</p>
        </div>
      </App>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the page title', () => {
    const title = 'Test title';

    const wrapper = shallow(
      <App title={title} showNavigation={false}>
        <div>
          <p>Test app children</p>
        </div>
      </App>
    );

    expect(wrapper.find('title').text()).toBe(title);
  });
});
