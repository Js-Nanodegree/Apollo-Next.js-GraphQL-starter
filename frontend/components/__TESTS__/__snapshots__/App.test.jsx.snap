// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snap shot 1`] = `
<div
  className="fullHeight"
>
  <main
    id="main"
  >
    <div>
      <p>
        Test app children
      </p>
    </div>
  </main>
</div>
`;
